src/components/heading/heading.scss

```diff
@@ -0,0 +1,125 @@
+@use '../../design-tokens/colors.yml';
+@use '../../design-tokens/layout.yml';
+@use "../../design-tokens/motion.yml";
+@use '../../design-tokens/opacity.yml';
+@use '../../design-tokens/sizes.yml';
+@use '../../mixins/focus';
+@use '../../mixins/headings';
+@use '../../mixins/ms';
+@use '../../mixins/unit';
+@use 'sass:math';
+
+$permalink-icon-size: sizes.$control-icon-size;
+$permalink-radius: sizes.$radius-full;
+$permalink-size: sizes.$control-height;
+
+$min-width-permalink-shift: layout.$max-width-prose + $permalink-size * 4;
+$max-width-permalink-shift: ($min-width-permalink-shift * 16 - 1) / 16;
+
+/**
+ * This containing element may be applied to a heading directly to apply a
+ * heading style, or to a containing element if you want to display a heading
+ * and permalink.
+ *
+ * 1. We default to the lowest heading level supported by HTML5, which forces
+ *    the template designer to specify an intended visual heading level.
+ * 2. Sets up the optional permalink to position itself relative to the
+ *    containing element.
+ */
+
+.c-heading {
+  @include headings.level(6); /* 1 */
+  position: relative; /* 2 */
+}
+
+/**
+ * Level modifiers
+ *
+ * These will apply the styles for a specific heading level.
+ *
+ * We start at `-2` so template designers have three sizes larger than a default
+ * `<h1>` to apply thoughtfully within templates.
+ *
+ * We end at 3 because that is the maximum depth we support with distinct
+ * visual differences.
+ */
+
+@for $level from -2 through headings.$max-level {
+  $level-segment: ms.step-class-segment($level);
+
+  .c-heading--level-#{$level-segment} {
+    @include headings.level($level);
+  }
+}
+
+/**
+ * If `c-heading` is applied to a containing element, this class must be applied
+ * to the inner heading to avoid unintended font sizes.
+ */
+
+.c-heading__element {
```

Comment:
I started with `__text` but changed it because, to me anyway, `text` suggests that HTML is not allowed. I changed it to `__element` because this is meant to be specifically the heading element, and `c-heading__heading-element` felt silly.

I'm open to `__content`, though. I also toyed with `__inner`.

Reply:
I'd be fine with `content` or `inner`. I think those communicate more intent than `element`

Reply:
- [x] Renamed to `c-heading__content`